docs/publish.md

```diff
@@ -0,0 +1,63 @@
+
+#### Debugging with Chrome Developer Tools
+
+Similar to the Chrome web browser, Mainframe OS provides developer tools ("devtools") to inspect the DOM, and view javascript console output (among other capabilities). However, unlike the web browser, there is not one set of devtools per app, but several sets of devtools. Most likely, you will want the app-specific devtools (see #1 below), but if you are developing the Mainframe OS stack itself, you may benefit from the others as well.
+
+1. The HTML, CSS, and javascript code specific to your app is associated with its own devtools window. When running Mainframe OS in developer mode, this window will open automatically when you open an app. You can also force it to open by running the following command from the app container ("Trusted UI") console:
+
+```
+$("#sandbox-webview").openDevTools();
+```
+
+2. Each app window has a "Trusted UI" container with associated devtools. Errors, warnings, and logs related to the "Trusted UI" portion of the app container will be visible there. Most people will not need this when developing an app. You can enable this view via the menu, when the app window has the focus: `View` -> `Toggle Developer Tools`.
+
+3. The main Launcher window also has its own developer tools pane. Errors, warnings, and logs associated with the main Launcher thread will be visible there. Most people will not need this when developing an app. You can enable this view via the menu, when the main launcher window has the focus: `View` -> `Toggle Developer Tools`.
+
+### Publishing your app
+
+To allow users to install your app, you will need to upload the contents to swarm and then write and sign a manifest file which can be distributed and used for installation.
+
+This can be done with the **Publish** button inside Mainframe OS, as described [here](introduction.md), if you installed Mainframe OS through the downloaded binaries.
+If you built Mainframe OS from source, the swarm-gateways upload limitation mentioned below will apply. Below are alternate steps to publish manually instead.
+
+#### Upload Content to Swarm
+
+By default, a vault is configured to connect to the [Swarm gateways](https://swarm-gateways.net/) to interact with Swarm. This works well for downloads and other requests, however uploads are limited to 1 MB, which might prevent from being able to upload some apps.
+In order to upload apps contents larger than 1 MB, a local Swarm node must be used. Installation instructions are available in the [Swarm guide](https://swarm-guide.readthedocs.io/en/latest/installation.html) and a [Docker image](https://github.com/ethersphere/swarm-docker) is also available.
+
+Your vault settings will need to be updated to use the local node rather than the Swarm gateways, running the following command (replace `http://localhost:8500` by the Swarm URL if different):
+
+```
+./packages/cli/bin/run vault:settings --bzz-url http://localhost:8500
```

Comment:
Do we still need to document this? I thought we set the default Swarm URL to our own gateway, without upload limitation?

Reply:
fixed.